index.js

```diff
@@ -1,41 +1,48 @@
 'use strict';
 
 const packagePath = 'node_modules/serverless-offline-direct-lambda';
-const handlerPath = `proxy.js`;
+const handlerPath = 'proxy.js';
 
 class ServerlessPlugin {
   constructor(serverless, options) {
     this.serverless = serverless;
     this.options = options;
 
     this.hooks = {
-      "before:offline:start:init": this.startHandler.bind(this),
+      'before:offline:start': this.startHandler.bind(this),
     };
   }
 
   startHandler() {
+    let location = '';
+    try {
+      location = this.serverless.service.custom['serverless-offline'].location;
+      this.serverless.service.custom['serverless-offline'].location = '';
+    } catch (_) { }
+
     this.serverless.cli.log('Running Serverless Offline with direct lambda support');
 
-    addProxies(this.serverless.service.functions);
+    addProxies(this.serverless.service.functions, location);
   }
 }
 
-const addProxies = functionsObject => {
+const addProxies = (functionsObject, location) => {
   Object.keys(functionsObject).forEach(fn => {
 
     // filter out functions with event config,
     // leaving just those intended for direct lambda-to-lambda invocation
     const functionObject = functionsObject[fn];
-    if (!functionObject.events || functionObject.events.length == 0) {
-      const pf = functionProxy(functionObject);
+    if (!functionObject.events || !functionObject.events.some((event) => event === 'http')) {
```

Comment:
I think this may be faulty - I just tried it locally and it created proxies for all functions, not just those with no http events (tested with serverless 1.27.3).

Reply:
You are right, it is indeed faulty. `event` is an object with `http` as key. I'll update the pull request with a fix.